crates/notion-core/src/catalog.rs

```diff
@@ -159,16 +200,46 @@ impl NodeCatalog {
 
     /// Resolves the specified semantic versioning requirements from the public distributor (`https://nodejs.org`).
     fn resolve_public(&self, matching: &VersionReq) -> Fallible<Installer> {
-        let spinner = progress_spinner(&format!(
-            "Fetching public registry: {}",
-            PUBLIC_NODE_VERSION_INDEX
-        ));
-        let serial: serial::index::Index = reqwest::get(PUBLIC_NODE_VERSION_INDEX)
-            .unknown()?
-            .json()
-            .unknown()?;
-        spinner.finish_and_clear();
-        let index = serial.into_index()?;
+        let index: Index = match read_cached().unknown()? {
+            Some(serial) => serial,
+            None => {
+                let spinner = progress_spinner(&format!(
+                    "Fetching public registry: {}",
+                    PUBLIC_NODE_VERSION_INDEX
+                ));
+                let mut response: reqwest::Response = reqwest::get(PUBLIC_NODE_VERSION_INDEX).unknown()?;
+                let response_text: String = response.text().unknown()?;
+                let cached: NamedTempFile = NamedTempFile::new().unknown()?;
+
+                // Block to borrow cached for cached_file.
+                {
+                    let mut cached_file: &File = cached.as_file();
+                    cached_file.write(response_text.as_bytes()).unknown()?;
+                }
+
+                cached.persist(path::node_index_file()?).unknown()?;
+
+                let expiry: NamedTempFile = NamedTempFile::new().unknown()?;
+
+                // Block to borrow expiry for expiry_file.
+                {
+                    let mut expiry_file: &File = expiry.as_file();
+                    let header: &reqwest::header::Raw = response.headers().get_raw("Expires").unwrap();
```

Comment:
@dherman — You briefly mentioned not liking the use of `.unwrap()` here, but we never got a chance to discuss alternatives.